src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -91,20 +92,46 @@ public void setupSchema(List<ColumnMetadata> columns) {
       if (!column.getNullable()) {
         addNonNullableFieldName(column.getInternalName());
       }
-      this.statsMap.put(
-          column.getInternalName(),
-          new RowBufferStats(column.getName(), column.getCollation(), column.getOrdinal()));
-
-      if (onErrorOption == OpenChannelRequest.OnErrorOption.ABORT
-          || onErrorOption == OpenChannelRequest.OnErrorOption.SKIP_BATCH) {
-        this.tempStatsMap.put(
+      if (!clientBufferParameters.getIsIcebergMode()) {
+        this.statsMap.put(
             column.getInternalName(),
-            new RowBufferStats(column.getName(), column.getCollation(), column.getOrdinal()));
+            new RowBufferStats(column.getName(), column.getCollation(), column.getOrdinal(), 0));
+
+        if (onErrorOption == OpenChannelRequest.OnErrorOption.ABORT
+            || onErrorOption == OpenChannelRequest.OnErrorOption.SKIP_BATCH) {
+          this.tempStatsMap.put(
+              column.getInternalName(),
+              new RowBufferStats(column.getName(), column.getCollation(), column.getOrdinal(), 0));
+        }
       }
 
       id++;
     }
     schema = new MessageType(PARQUET_MESSAGE_TYPE_NAME, parquetTypes);
+    if (clientBufferParameters.getIsIcebergMode()) {
+      int ordinal = 0;
+      String prevParentColumnName = "";
+      for (ColumnDescriptor columnDescriptor : schema.getColumns()) {
+        String parentColumnName = columnDescriptor.getPath()[0];
+        if (!parentColumnName.equals(prevParentColumnName)) {
+          ordinal++;
+          prevParentColumnName = parentColumnName;
+        }
+        String subColumnName = String.join(".", columnDescriptor.getPath());
+        int fieldId =
+            parentColumnName.equals(subColumnName)
+                ? 0
+                : columnDescriptor.getPrimitiveType().getId().intValue();
+        RowBufferStats stats = new RowBufferStats(subColumnName, null, ordinal, fieldId);
```

Comment:
hardcoding null for the collation - lets chat about this, leaving a comment so i don't forget!